### PR TITLE
feat(frontend/backend/shared): Search courses tm-55

### PR DESCRIPTION
**Pull request by GEOFARL**

### CODE REVIEW

File: apps/frontend/src/libs/components/search-bar/search-bar.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import styles from "./styles.module.css";
+
+const SearchBar: React.FC = () => {
+	const { user } = useAppSelector((state) => ({
+		user: state.auth.user as UserAuthResponseDto,
+	}));
+	const dispatch = useAppDispatch();
+
+	const { control, errors, handleSubmit } = useAppForm({
+		defaultValues: DEFAULT_SEARCH_COURSE_PAYLOAD,
```
Comment (tarmiel) [inline apps/frontend/src/libs/components/search-bar/search-bar.tsx:26]: You do not need vendors in this form

File: apps/frontend/src/pages/overview/libs/constants/constants.ts [hunk 2]
```diff
@@ -1,13 +1,4 @@
...
-	vendors: {},
-};
-
-const SEARCH_COURSES_DELAY_MS = 700;
-
-export { DEFAULT_SEARCH_COURSE_PAYLOAD, SEARCH_COURSES_DELAY_MS };
+export {
+	DEFAULT_SEARCH_COURSE_PAYLOAD,
+	SEARCH_COURSES_DELAY_MS,
+} from "~/modules/courses/courses.js";
```
Comment (tarmiel) [inline apps/frontend/src/pages/overview/libs/constants/constants.ts:4]: Why did you need to move this to courses module?

File: apps/frontend/src/libs/components/search-bar/search-bar.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+				className={styles["search"]}
+				color="light"
+				control={control}
+				errors={errors}
+				hasVisuallyHiddenLabel
+				label="Search"
+				name="search"
+				placeholder="Search or type"
+			/>
+			<Icon className={styles["search-icon"]} name="search" />
```
Comment (tarmiel) [inline apps/frontend/src/libs/components/search-bar/search-bar.tsx:62]: I think icon needs to be part of input component

File: apps/frontend/src/pages/overview/libs/constants/constants.ts (hunk 2 above)
Comment (GEOFARL) [reply apps/frontend/src/pages/overview/libs/constants/constants.ts:4]: SEARCH_COURSES_DELAY_MS I need to be able to use debounce as you do in the modal window, I'll return DEFAULT_SEARCH_COURSE_PAYLOAD

File: apps/backend/src/modules/courses/course.repository.ts [hunk 4]
```diff
@@ -234,6 +234,49 @@ class CourseRepository implements Repository<CourseEntity> {
 		);
 	}
 
+	public async findByUserIdWithSearch(
```
Comment (fshabanov) [inline apps/backend/src/modules/courses/course.repository.ts:237]: Should be a part of `findByUserId`, no need to create a separate method

File: apps/frontend/src/modules/user-courses/slices/actions.ts [hunk 5]
```diff
@@ -29,7 +29,23 @@ const loadMyCourses = createAsyncThunk<
...
 
-	return userCourseApi.getAllByUserId(userId);
+	return userCourseApi.getAllByUserId({
+		id: userId,
+		search: "",
+	});
 });
 
-export { add, loadMyCourses };
+const searchMyCourses = createAsyncThunk<
```
Comment (fshabanov) [inline apps/frontend/src/modules/user-courses/slices/actions.ts:38]: Use the existing action

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by MappariChan — APPROVED** [review]

File: apps/backend/src/modules/courses/course.repository.ts [hunk 6]
```diff
@@ -208,6 +214,7 @@ class CourseRepository implements Repository<CourseEntity> {
 		const courseModels = await user
 			.$relatedQuery(DatabaseTableName.COURSES)
 			.for(userId)
+			.where("title", "ilike", `%${search}%`)
```
Comment (Oleksandra-Masiuk) [inline apps/backend/src/modules/courses/course.repository.ts:217]: Please, use [whereILike](https://knexjs.org/guide/query-builder.html#whereilike)

File: apps/frontend/src/libs/components/input/input.tsx [hunk 7]
```diff
@@ -28,12 +31,16 @@ const Input = <T extends FieldValues>({
...
 	errors,
 	hasVisuallyHiddenLabel,
+	iconName,
 	label,
 	name,
 	placeholder = "",
 	type = "text",
 }: Properties<T>): JSX.Element => {
 	const { field } = useFormController({ control, name });
+	const icon = iconName ? (
```
Comment (Oleksandra-Masiuk) [inline apps/frontend/src/libs/components/input/input.tsx:41]: add `const hasIcon  = Boolean(iconName);` and use it here

File: apps/frontend/src/libs/components/input/input.tsx [hunk 8]
```diff
@@ -43,6 +50,7 @@ const Input = <T extends FieldValues>({
 		styles["input"],
 		styles[color],
 		hasError && styles["error-input"],
+		icon && styles["input-with-icon"],
```
Comment (Oleksandra-Masiuk) [inline apps/frontend/src/libs/components/input/input.tsx:53]:
```suggestion
		hasIcon && styles["input-with-icon"],
```

**Review by Oleksandra-Masiuk — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**